File: backend/app/schemas/jobs.py
from typing import Any, Dict, Optional

from pydantic import BaseModel, Field


class JobCreate(BaseModel):
    type: str = Field(..., min_length=1)
    payload: Dict[str, Any] = Field(default_factory=dict)


class JobResponse(BaseModel):
    job_id: str
    status: str


class JobRecord(BaseModel):
    job_id: str
    type: str
    status: str
    created_at: str
    updated_at: str
    progress: Optional[float] = None
    message: Optional[str] = None
    payload: Dict[str, Any] = Field(default_factory=dict)
    result: Optional[Dict[str, Any]] = None
    error: Optional[Dict[str, Any]] = None

File: backend-bun/src/schemas/jobs.ts
import { z } from "zod";

// Request schemas
export const JobCreateSchema = z.object({
  type: z.string().min(1),
  payload: z.record(z.string(), z.unknown()).optional(),
});

export type JobCreateInput = z.infer<typeof JobCreateSchema>;

// Response types
export interface JobResponse {
  job_id: string;
  status: string;
}

export interface JobRecord {
  job_id: string;
  type: string;
  status: string;
  created_at: string;
  updated_at: string;
  progress: number | null;
  message: string | null;
  payload: Record<string, unknown>;
  result: Record<string, unknown> | null;
  error: Record<string, unknown> | null;
}

// Job status enum
export const JobStatus = {
  QUEUED: "queued",
  RUNNING: "running",
  SUCCEEDED: "succeeded",
  FAILED: "failed",
  CANCELED: "canceled",
} as const;

export type JobStatusType = (typeof JobStatus)[keyof typeof JobStatus];
